hist/histv7/test/axis.cxx

```diff
@@ -3,183 +3,540 @@
-
-    EXPECT_LT(ax.GetBinCenter(0), -1.);
-    EXPECT_LT(ax.GetBinFrom(0), -1.);
-    EXPECT_FLOAT_EQ(-1., ax.GetBinTo(0));
-
-    EXPECT_LT(ax.GetBinCenter(-1), -1.);
-    EXPECT_LT(ax.GetBinFrom(-2), -1.);
-    EXPECT_LE(ax.GetBinTo(-3), -1.);
-
-    EXPECT_LT(1., ax.GetBinCenter(11));
-    EXPECT_FLOAT_EQ(1., ax.GetBinFrom(11));
-    EXPECT_LT(1., ax.GetBinTo(11));
-
-    EXPECT_LT(1., ax.GetBinCenter(111));
-    EXPECT_LE(1., ax.GetBinFrom(111));
-    EXPECT_LT(1., ax.GetBinTo(111));
-    
-
-    RAxisConfig axcfg(ax);
-    EXPECT_EQ(ax.GetNBins(), axcfg.GetNBins());
-    EXPECT_EQ(10, axcfg.GetNBinsNoOver());
-    EXPECT_EQ(0, axcfg.GetUnderflowBin());
-    EXPECT_EQ(11, axcfg.GetOverflowBin());
-    EXPECT_EQ(true, axcfg.IsUnderflowBin(0));
-    EXPECT_EQ(true, axcfg.IsUnderflowBin(-10000));
-    EXPECT_EQ(true, axcfg.IsOverflowBin(11));
-    EXPECT_EQ(true, axcfg.IsOverflowBin(16));
-  }
-
-  {
-    RAxisGrow ax(10, -1., 1.);
-    EXPECT_EQ(10, ax.GetNBins());
-    EXPECT_EQ(10, ax.GetNBinsNoOver());
-    EXPECT_EQ(0, ax.GetUnderflowBin());
-    EXPECT_EQ(11, ax.GetOverflowBin());
-    EXPECT_EQ(true, ax.IsUnderflowBin(0));
-    EXPECT_EQ(true, ax.IsUnderflowBin(-1));
-    EXPECT_EQ(true, ax.IsOverflowBin(11));
-    EXPECT_EQ(true, ax.IsOverflowBin(12));
-
-    RAxisConfig axcfg(ax);
-    EXPECT_EQ(ax.GetNBins(), axcfg.GetNBins());
-    EXPECT_EQ(10, axcfg.GetNBinsNoOver());
-    EXPECT_EQ(0, axcfg.GetUnderflowBin());
-    EXPECT_EQ(11, axcfg.GetOverflowBin());
-    EXPECT_EQ(true, axcfg.IsUnderflowBin(0));
-    EXPECT_EQ(true, axcfg.IsUnderflowBin(-1));
-    EXPECT_EQ(true, axcfg.IsOverflowBin(11));
-    EXPECT_EQ(true, axcfg.IsOverflowBin(12));
-  }
-
-  {
-    RAxisIrregular ax("RITLE", {-0.1, 0.2, 0.5, 10.});
-    EXPECT_EQ(3 + nOverflow, ax.GetNBins());
-    EXPECT_EQ(3, ax.GetNBinsNoOver());
-    EXPECT_EQ(0, ax.GetUnderflowBin());
-    EXPECT_EQ(4, ax.GetOverflowBin());
-    EXPECT_EQ(true, ax.IsUnderflowBin(0));
-    EXPECT_EQ(true, ax.IsUnderflowBin(-1));
-    EXPECT_EQ(true, ax.IsOverflowBin(4));
-    EXPECT_EQ(true, ax.IsOverflowBin(5000));
-
-    RAxisConfig axcfg(ax);
-    EXPECT_EQ(ax.GetNBins(), axcfg.GetNBins());
-    EXPECT_EQ(3, axcfg.GetNBinsNoOver());
-    EXPECT_EQ(0, axcfg.GetUnderflowBin());
-    EXPECT_EQ(4, axcfg.GetOverflowBin());
-    EXPECT_EQ(true, axcfg.IsUnderflowBin(0));
-    EXPECT_EQ(true, axcfg.IsUnderflowBin(-1));
-    EXPECT_EQ(true, axcfg.IsOverflowBin(5));
-    EXPECT_EQ(true, axcfg.IsOverflowBin(4));
+// FIXME: Passing this directly as an initializer list is ambiguous because
+//        the compiler doesn't know if it should convert the inner const char*
+//        literals to std::string_view or std::string.
+std::vector<std::string_view> labels{"abc", "de", "fghi", "j", "klmno"};
+
+// Test RAxisConfig and conversion to concrete axis types
+TEST(AxisTest, Config) {
+  // Equidistant
+  {
+    auto test = [](const RAxisConfig& cfg, std::string_view title) {
+      EXPECT_EQ(cfg.GetTitle(), title);
```

Comment:
I can't believe I got actual vs expected wrong, i.e. the gtest order of arguments :-( Thanks for fixing this!

Reply:
Totally not on purpose, I didn't even notice that gtest differentiated its lhs from rhs ;)